core/domain/classifier_services.py

```diff
@@ -124,6 +124,57 @@ def classify_string_classifier_rule(state, normalized_answer):
 
     return None
 
+def train(exploration):
```

Comment:
I'm not really sure about how this would work when it's inside save_exploration(). I think the creator ends up waiting for training to finish (which is a lot of time, not good). 
@seanlip Haven't worked with asynchronous processes, could you point me to an example in Oppia codebase, if there's one?
And, there's a follow up PR to modify something here. Currently, the train() happens every time a creator modifies something (even a minor change). The only check is if the state is trainable. We are adding another check to see if the changes made classify under the re-training conditions we mentioned in the doc. That will ensure that the classifier isn't re-trained until necessary (unlike right now).